### PR TITLE
Simplify publishing of documentation

### LINKED ISSUE #11152: Documentation for our many products
Hi.

I have fantastic news. We are growing! 🐈 Together with this, our documentation grows and we have newer needs and requirements. 

We currently only publish documentation for our app releases. 
 - https://airflow.apache.org/docs/stable/

But our community is also working on other products that have a different release cycle, use case, docs build tool and needs
- Production Docker image: https://hub.docker.com/r/apache/airflow
- REST API: https://airflow.readthedocs.io/en/latest/stable-rest-api-ref.html (it has a separate version number, so a version index could be helpful. We can also set the version to the application version)
- Backport/provider packages: https://github.com/apache/airflow/tree/master/backport_packages
- Helm Chart: https://github.com/apache/airflow/tree/master/chart
- [API Clients for Apache Airflow](https://github.com/apache/airflow/issues/10176)
  - Go: https://github.com/apache/airflow-client-go 
  - Python: https://github.com/apache/airflow-client-python
  - Java: https://github.com/apache/airflow-client-java
  - Javascript: https://github.com/apache/airflow-client-javascript
- Apache Airflow on K8S Operator: https://github.com/apache/airflow-on-k8s-operator

and other: https://github.com/apache/airflow/issues/10550, https://github.com/apache/airflow/issues/10552

I think it is worth considering how we want to inform users about these products on our website.  

One solution is to create an index at: `https://airflow.apache.org/docs/` (currently there is a redirection to the latest stable version of the documentation), which will refer you to separate documentation for each product.
It will look similar to the examples:
https://mc-stan.org/
https://cloud.google.com/sql/docs/
https://www.terraform.io/docs/

An alternative solution is moving all documentations to our app documentation:
https://github.com/apache/airflow/pull/11136
https://github.com/apache/airflow/pull/10998

What is our plan for this documentation? Do we need to standardize our documentation building tools? What requirements does each documentation have, e.g. do we always need versioning? 
CC: @kaxil @potiuk

### LINKED ISSUE #11423: Separate out documentation building and publishing per provider

### PR DESCRIPTION
**Pull request by mik-laj**
Close: https://github.com/apache/airflow/issues/11423
Close: https://github.com/apache/airflow/issues/11152

Depends on: https://github.com/apache/airflow-site/pull/318

This is a small automation to make publishing documentation for multiple packages easier.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: dev/README_RELEASE_AIRFLOW.md [hunk 1]
```diff
@@ -551,6 +552,43 @@ At this point we release an official package:
 
 - Update CHANGELOG.md with the details, and commit it.
 
+## Publish documentation
+
+Documentation is an essential part of the product and should be made available to users.
+In our cases, documentation  for the released versions is published in a separate repository - [`apache/airflow-site`](https://github.com/apache/airflow-site), but the documentation source code and build tools are available in the `apache/airflow` repository, so you have to coordinate between the two repositories to be able to build the documentation.
```
Comment (kaxil) [inline dev/README_RELEASE_AIRFLOW.md]:
```suggestion
Documentation is an essential part of the product and should be made available to users.
In our cases, documentation for the released versions is published in a separate repository - [`apache/airflow-site`](https://github.com/apache/airflow-site), but the documentation source code and build tools are available in the `apache/airflow` repository, so you have to coordinate between the two repositories to be able to build the documentation.
```

File: dev/README_RELEASE_AIRFLOW.md [hunk 2]
```diff
@@ -551,6 +552,43 @@ At this point we release an official package:
...
+    ```shell script
+    cd "${AIRFLOW_REPO_ROOT}"
+    ./breeze build-docs -- --package apache-airflow --for-production
+    ```
+
+- Now you can preview the documentation.
+
+    ```shell script
+    ./docs/start_doc_server.sh
+    ``
```
Comment (kaxil) [inline dev/README_RELEASE_AIRFLOW.md]:
```suggestion
    ```
```

File: dev/README_RELEASE_AIRFLOW.md [hunk 3]
```diff
@@ -551,6 +552,43 @@ At this point we release an official package:
...
+
+- Copy the documentation to the ``airflow-site`` repository, create commit and push changes.
+
+    ```shell script
+    ./docs/publish_docs.py --package apache-airflow
+    cd "${AIRFLOW_SITE_DIRECTORY}"
+    git commit -m "Add documentation for Apache Airflow ${VERSION}"
+    git push
+    ````
```
Comment (kaxil) [inline dev/README_RELEASE_AIRFLOW.md]:
```suggestion
    ```
```

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 4]
```diff
@@ -884,6 +885,58 @@ twine upload -r pypi dist/*
...
+      --package apache-airflow-providers-apache-airflow \
+      --package apache-airflow-providers-telegram \
+      --for-production
+    ```
+
+- Now you can preview the documentation.
+
+    ```shell script
+    ./docs/start_doc_server.sh
+    ``
```
Comment (kaxil) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]:
```suggestion
    ```
```

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 5]
```diff
@@ -884,6 +885,58 @@ twine upload -r pypi dist/*
...
+- Copy the documentation to the ``airflow-site`` repository
+
+    ```shell script
+    ./docs/publish_docs.py \
+        --package apache-airflow-providers \
+        --package apache-airflow-providers-apache-airflow \
+        --package apache-airflow-providers-telegram \
+
+    cd "${AIRFLOW_SITE_DIRECTORY}"
+    ````
```
Comment (kaxil) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]:
```suggestion
    ```
```

**Review by kaxil — APPROVED** [review]
Minor changes requested

**Comment by kaxil** [comment]
ping @mik-laj This PR is good to go -- just few minor suggestions.

File: dev/README_RELEASE_AIRFLOW.md [hunk 6]
```diff
@@ -583,7 +621,7 @@ https://pypi.python.org/pypi/apache-airflow
 
 The documentation is available on:
 https://airflow.apache.org/
-https://airflow.apache.org/docs/${VERSION}/
+https://airflow.apache.org/docs/apache-airflow/${VERSION}/
```
Comment (ashb) [inline dev/README_RELEASE_AIRFLOW.md:624]: Something about having `airflow` appear in the URL twice bugs me unduly, but I don't really have an alternate suggestion (and I agree that having providers separately is right)

Oh well.

**Comment by kaxil** [comment]
Should we merge https://github.com/apache/airflow-site/pull/318 and this one @mik-laj

**Comment by mik-laj** [comment]
@kaxil We can merge this PR now.  https://github.com/apache/airflow-site/pull/318 should be merged with the first release of the documentation, as there is a new layour for the /docs/ page.

File: dev/README_RELEASE_AIRFLOW.md (hunk 6 above)
Comment (mik-laj) [reply dev/README_RELEASE_AIRFLOW.md:624]: I also have no idea for a better directory structure. Now we will be able to create a documentation link easily based on the package name in the pypi repository.

```
https://airflow.apache.org/docs/${PYPI-NAME}/${VERSION}/
```

**Pull request merged**